### PR TITLE
Refine scalarizeGenericInput

### PR DESCRIPTION
**Pull request by xuechen417**
For non-constant index of ExtractElementInst, we just skip it in
`PatchResourceCollect::scalarizeGenericInput`. There is no need assign
the array `elementUsed` all false as in commit 1dda2ad.

### CODE REVIEW

File: llpc/test/shaderdb/PipelineVsFs_TestInOutPacking.pipe [hunk 1]
```diff
@@ -1,6 +1,75 @@
 ; BEGIN_SHADERTEST
 ; RUN: amdllpc -spvgen-dir=%spvgendir% -v %gfxip %s | FileCheck -check-prefix=SHADERTEST %s
-; SHADERTEST-LABEL: {{^// LLPC}} SPIRV-to-LLVM translation results
+; SHADERTEST-LABEL: {{^// LLPC}} location input/output mapping results (FS shader)
+; CHECK:      (FS) Input:  loc = 0, comp = 0 =>  Mapped = 0, 0
```
Comment (kuhar) [inline llpc/test/shaderdb/PipelineVsFs_TestInOutPacking.pipe:4]: Do lines with `CHECK` do anything if FileCheck is invoked with `-check-prefix=SHADERTEST`?

**Review by kuhar — CHANGES_REQUESTED** [review]
I'm not sure if this is the best way to test it. To me it seems like we can tell that it does not crash, but not much more than that. Would it be difficult to create a new test that tries to exercise only the case when `extractelement`'s operand is dynamic?

File: llpc/test/shaderdb/PipelineVsFs_TestInOutPacking.pipe (hunk 1 above)
Comment (xuechen417) [reply llpc/test/shaderdb/PipelineVsFs_TestInOutPacking.pipe:4]: You are right. I will add a new test to check this corner case.

**Comment by trenouf** [comment]
Also, could you summarize the change in the summary line please? If you do a "git log --oneline", the summary "Refine commit 1dda2ad" does not say what the change is. Thanks.

**Comment by xuechen417** [comment]
> Also, could you summarize the change in the summary line please? If you do a "git log --oneline", the summary "Refine commit [1dda2ad](https://github.com/GPUOpen-Drivers/llpc/commit/1dda2ad4deb37b6f37cac23e3bc60aceaaa0f16b)" does not say what the change is. Thanks.

Got it. Thanks.

**Comment by xuechen417** [comment]
V1: Add a lgc test for the corner case.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 5a833fa839df71df9555fa0ea3c8de8fd42db5be
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 1311c5cdfe6cde2faa349de66b2e09cbf49306aa</li>
<li>SPVGEN: dev bb3da8bc00bcf4373d563bf9b3c74a80a8b28fe6</li>
<li>XGL: dev 5dbd470d8c1cbf8ee5d9a1204ba23a2fbf88a242</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 4f968af13313fd60f4a661943745d252ad14f2af</li>
</ul></details>

<details><summary>CTS tests (Failed: 5/40952)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 6367/10238 (62.2%)</li>
<li><details><summary>Failed: 1/10238 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.pipeline.dynamic_offset.compute.single_set.uniform_buffer.numcmdbuffers_1.sameorder.numdescriptorsetbindings_1.numdynamicbindings_1.numnondynamicbindings_0
Stack trace: Script:
Crash

```
</p></details></li>

<li>Not Supported: 3870/10238 (37.8%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 6366/10238 (62.2%)</li>
<li><details><summary>Failed: 2/10238 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.memory.pipeline_barrier.all.1048576_vertex_buffer_stride_4
Stack trace: Script:
10:HostMemoryAccess Result differs from reference, Expected: 0x00000064, Got: 0x000000c9, At offset: 289742


FAILURE: dEQP-VK.pipeline.dynamic_offset.compute.single_set.uniform_buffer.numcmdbuffers_1.sameorder.numdescriptorsetbindings_1.numdynamicbindings_1.numnondynamicbindings_0
Stack trace: Script:
Crash

```
</p></details></li>

<li>Not Supported: 3870/10238 (37.8%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 6367/10238 (62.2%)</li>
<li><details><summary>Failed: 1/10238 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.pipeline.dynamic_offset.compute.single_set.uniform_buffer.numcmdbuffers_1.sameorder.numdescriptorsetbindings_1.numdynamicbindings_1.numnondynamicbindings_0
Stack trace: Script:
Crash

```
</p></details></li>

<li>Not Supported: 3870/10238 (37.8%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 6396/10238 (62.5%)</li>
<li><details><summary>Failed: 1/10238 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.pipeline.dynamic_offset.compute.single_set.uniform_buffer.numcmdbuffers_1.sameorder.numdescriptorsetbindings_1.numdynamicbindings_1.numnondynamicbindings_0
Stack trace: Script:
Crash

```
</p></details></li>

<li>Not Supported: 3841/10238 (37.5%)</li>
</ul></details>


</ul></details>

**Comment by xuechen417** [comment]
V2: Rebase

**Review by kuhar — APPROVED** [review]

**Review by amdrexu — APPROVED** [review]
